tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,108 @@
+import os
+from pathlib import Path
+from unittest.case import TestCase
+
+import boto3
+import pytest
+import yaml
+from samcli.lib.deploy.deployer import Deployer
+from tests_integ.helpers.helpers import transform_template, verify_stack_resources, generate_suffix, create_bucket
+
+STACK_NAME_PREFIX = "sam-integ-stack-"
+S3_BUCKET_PREFIX = "sam-integ-bucket-"
+CODE_KEY_TO_FILE_MAP = {"codeuri": "code.zip", "contenturi": "layer1.zip", "definitionuri": "swagger1.json"}
+
+
+class BaseTest(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        BaseTest.tests_integ_dir = Path(__file__).resolve().parents[1]
+        BaseTest.resources_dir = Path(BaseTest.tests_integ_dir, "resources")
+        BaseTest.template_dir = Path(BaseTest.resources_dir, "templates", "single")
+        BaseTest.output_dir = BaseTest.tests_integ_dir
+        BaseTest.expected_dir = Path(BaseTest.resources_dir, "expected", "single")
+        code_dir = Path(BaseTest.resources_dir, "code")
+
+        BaseTest.s3_bucket_name = S3_BUCKET_PREFIX + generate_suffix()
+        session = boto3.session.Session()
+        my_region = session.region_name
+        create_bucket(BaseTest.s3_bucket_name, region=my_region)
+
+        s3_client = boto3.client("s3")
+        BaseTest.code_key_to_url = {}
+
+        for key, file_name in CODE_KEY_TO_FILE_MAP.items():
+            code_path = str(Path(code_dir, file_name))
+            s3_client.upload_file(code_path, BaseTest.s3_bucket_name, file_name)
+            code_url = f"s3://{BaseTest.s3_bucket_name}/{file_name}"
+            BaseTest.code_key_to_url[key] = code_url
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        BaseTest._clean_bucket()
+
+    @classmethod
+    def _clean_bucket(cls):
+        s3_client = boto3.client("s3")
+        response = s3_client.list_objects(Bucket=BaseTest.s3_bucket_name)
+        for content in response["Contents"]:
```

Comment:
This is possibly okay because of the fixed number of resources we add, but if this request is paginated you will not properly delete all bucket objects. However, the [S3 Bucket Resource](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Bucket.objects) has methods that can iterate over all objects and handle pagination automatically, which is a better choice for this method.

Reply:
The list_objects will return 1000 objects per page, I'm not sure we'll ever reach this amount?
Also, we should maybe use list_objects_v2 instead.

Reply:
I test the bucket.objects.all() method in [S3 Bucket Resource](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Bucket.objects), this one should work without the constraint of paginated.

Reply:
It seems to fetch all the objects at once and then paginate on the client which, performance wise, isn't ideal.

Reply:
Alex confirmed objects.all() streams pages of 1000 elements and doesn't load everything at once.
Good for me!